src/include/rbtree.h

```diff
@@ -1,17 +1,173 @@
+/*
+ * The primary function of red-black trees is to provide a fast (O(ln(n))) and cost-effective method for mapping address ranges to numbers and accessing them by specifying an address within that range.
+ * Each RB-node includes two additional pointers, "start" and "end," to denote a memory range.
+ * When a range of memory is mapped from the system, it is recorded by multiple red-black trees with the information of that memory range (e.g., permissions).
+ * That's it! Each node encapsulates a memory range along with detailed information about that range.
+ * Before the introduction of the red-black tree in Box64, 
+ * the rationale for memory management was a sparse array, 
+ * which take O(n) complexity for accessing data.
+ * After transitioning from a sparse array to a red-black tree, 
+ * the memory usage has decreased slightly for processes that consumed a lot of RAM 
+ * (for example, Steam uses about 100 MB less memory, and each Wine process uses about 15 MB less).
```

Comment:
This is really weird to read: the beginning has no break mid-sentence, then midway through there are line breaks after every comma. Please either always add breaks or keep every sentence in one line.
```suggestion
 * Before the introduction of the red-black tree in Box64, the rationale for memory management was a sparse array, which take O(n) complexity for accessing data.
 * After transitioning from a sparse array to a red-black tree, the memory usage has decreased slightly for processes that consumed a lot of RAM (for example, Steam uses about 100 MB less memory, and each Wine process uses about 15 MB less).
```